state-sync/inter-component/consensus-notifications/src/lib.rs

```diff
@@ -0,0 +1,424 @@
+        });
+
+        // Give the thread enough time to spawn and send the notification
+        std::thread::sleep(Duration::from_millis(1000));
+
+        // Verify the notification arrives at the receiver
+        match consensus_listener.notification_receiver.try_next() {
+            Ok(Some(consensus_notification)) => match consensus_notification {
+                ConsensusNotification::SyncToTarget(sync_notification) => {
+                    assert_eq!(create_ledger_info(), sync_notification.target);
+                }
+                result => panic!("Expected consensus sync notification but got: {:?}", result),
+            },
+            result => panic!("Expected consensus notification but got: {:?}", result),
+        };
+    }
+
+    #[test]
+    fn test_consensus_notification_responses() {
+        // Create runtime and consensus notifier
+        let runtime = create_runtime();
+        let _enter = runtime.enter();
+        let (consensus_notifier, mut consensus_listener) = ConsensusNotifier::new(1000);
+
+        // Spawn a new thread to handle any messages on the receiver
+        let _handler = std::thread::spawn(move || loop {
+            match consensus_listener.notification_receiver.try_next() {
+                Ok(Some(ConsensusNotification::NotifyCommit(commit_notification))) => {
+                    let _result = block_on(
+                        consensus_listener
+                            .respond_to_commit_notification(commit_notification, Ok(())),
+                    );
+                }
+                Ok(Some(ConsensusNotification::SyncToTarget(sync_notification))) => {
+                    let _result = block_on(consensus_listener.respond_to_sync_notification(
+                        sync_notification,
+                        Err(Error::UnexpectedErrorEncountered("Oops?".into())),
+                    ));
+                }
+                _ => { /* Do nothing */ }
+            }
+        });
+
+        // Send a commit notification and verify a successful response
+        let notify_result =
+            block_on(consensus_notifier.notify_new_commit(vec![create_user_transaction()], vec![]));
+        assert!(notify_result.is_ok());
+
+        // Send a sync notification and very an error response
+        let notify_result = block_on(consensus_notifier.sync_to_target(create_ledger_info()));
+        assert!(notify_result.is_err());
+    }
+
+    fn create_user_transaction() -> Transaction {
+        let private_key = Ed25519PrivateKey::generate_for_testing();
+        let public_key = private_key.public_key();
+
+        let transaction_payload = TransactionPayload::Script(Script::new(vec![], vec![], vec![]));
+        let raw_transaction = RawTransaction::new(
+            AccountAddress::random(),
+            0,
+            transaction_payload,
+            0,
+            0,
+            "".into(),
+            0,
+            ChainId::new(10),
+        );
```

Comment:
We should consider sometime making dummy `RawTransaction` a test function on `RawTransaction`, mostly cause I know this is a dummy, but who knows what any of these inputs are :)

Reply:
Yeah.. there's a few of these core structs that tend to have very complicated `new` calls for test purposes. I'll leave this for now, but mark it down on my "potential cleanups" pile